backend/src/main/java/com/rcbsukcesja/hack2react/service/MessageService.java

```diff
@@ -21,48 +26,90 @@ public class MessageService {
 
     @Transactional
-    public MessageView createMessage(MessageDto messageDto) {
-        Message message = messageMapper.messageDtoToMessage(messageDto);
-        User fromUser = userService.getUserByIdOrThrowException(messageDto.fromUserId());
-        User toUser = userService.getUserByIdOrThrowException(messageDto.toUserId());
-        message.setFromUser(fromUser);
-        message.setToUser(toUser);
-        message.setConversation(conversationService
-                .getConversationByIdOrThrowException(messageDto.conversationId()));
+    public MessageView createMessage(MessageSaveDto messageSaveDto) {
+
+        Message message = new Message();
+        setBasicMessageFields(message, messageSaveDto);
+
+        User senders = getLogedUser();
+        message.setSender(senders);
+        User receiver = userService.getUserByIdOrThrowException(messageSaveDto.receiverId());
+        message.setReceiver(receiver);
         message.setCreatedAt(TimeUtils.nowInUTC());
-        Message saved = messageRepository.save(message);
-        return messageMapper.messageToMessageView(saved);
+        message.setName(createName(senders, messageSaveDto.name()));
 
+        messageRepository.save(message);
+        return messageMapper.messageToMessageView(message);
     }
 
-    @Transactional
-    public MessageView updateMessage(UUID messageId, MessageDto messageDto) {
+    private String createName(User user, String tempName) {
+
+        StringBuilder sB = new StringBuilder();
+
+        if (user.getUserType().equals(UserType.CITY_HALL)) {
+            sB.append(tempName).append(" ").append(UMK);
+        } else {
+            sB.append(tempName).append(" ").append(user.getOrganization().getName());
+        }
+        return sB.toString();
+    }
+
+
+    private void setBasicMessageFields(Message message, MessageSaveDto messageSaveDto) {
+        message.setMessage(messageSaveDto.message());
+        message.setTitle(messageSaveDto.title());
+        message.setContact(messageSaveDto.contact());
+    }
+
+    public MessageView patchUpdateMessage(UUID messageId, MessagePathDto dto) {
+
         Message actual = getMessageOrThrowException(messageId);
-        actual.setConversation(conversationService
-                .getConversationByIdOrThrowException(messageDto.conversationId()));
-        actual.setText(messageDto.text());
 
-        Message saved = messageRepository.save(actual);
-        return messageMapper.messageToMessageView(saved);
+        if (dto.contact() != null && !actual.getContact().equals(dto.contact())) {
+            actual.setContact(dto.contact());
+        }
+        if (dto.message() != null && !actual.getMessage().equals(dto.message())) {
+            actual.setMessage(dto.message());
+        }
+        if (dto.contact() != null && !actual.getName().equals(dto.name())) {
+            actual.setMessage(dto.message());
+        }
+        if (dto.name() != null && !actual.getName().split(" ")[0].equals(dto.name())) {
```

Comment:
A co jak ktoś da spację w namie? :)
```suggestion
        if (dto.name() != null) {
```